src/wire/ipv6fragment.rs

```diff
@@ -0,0 +1,280 @@
+    }
+
+    /// Set the fragment offset field.
+    #[inline]
+    pub fn set_frag_offset(&mut self, value: u16) {
+        let data = self.buffer.as_mut();
+        // Retain the lower order 3 bits
+        let raw = ((value & 0x1fff) << 3) | ((data[3] & 0x3) as u16);
+        NetworkEndian::write_u16(&mut data[field::FR_OF_M], raw);
+    }
+
+    /// Set 2-bit reserved field between fragment offset and more fragments.
+    #[inline]
+    pub fn set_res(&mut self, value: u8) {
+        let data = self.buffer.as_mut();
+        // Retain the higher order 5 bits and lower order 1 bit
+        let raw = (value & 0x6) | (data[3] & 0xf9);
+        data[3] = raw;
+    }
+
+    /// Set the more fragments flag field.
+    #[inline]
+    pub fn set_more_frags(&mut self, value: bool) {
+        let data = self.buffer.as_mut();
+        // Retain the high order 7 bits
+        let raw = (data[3] & 0xfe) | (value as u8 & 0x1);
+        data[3] = raw;
+    }
+
+    /// Set the fragmentation identification field.
+    #[inline]
+    pub fn set_ident(&mut self, value: u32) {
+        let data = self.buffer.as_mut();
+        NetworkEndian::write_u32(&mut data[field::IDENT], value);
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> fmt::Display for Header<&'a T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match Repr::parse(self) {
+            Ok(repr) => write!(f, "{}", repr),
+            Err(err) => {
+                write!(f, "IPv6 Fragment ({})", err)?;
+                Ok(())
+            }
+        }
+    }
+}
+
+/// A high-level representation of an IPv6 Fragment header.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct Repr {
+    /// The type of header immediately following the Fragment header.
+    pub next_header: Protocol,
+    /// The offset of the data following this header, relative to the start of the Fragmentable
+    /// Part of the original packet.
+    pub frag_offset: u16,
+    /// Whethere are not there are more fragments following this header
+    pub more_frags: bool,
+    /// The identification for every packet that is fragmented.
+    pub ident: u32,
+
+}
+
+impl Repr {
+    /// Parse an IPv6 Fragment Header and return a high-level representation.
+    pub fn parse<T>(header: &Header<&T>) -> Result<Repr> where T: AsRef<[u8]> + ?Sized {
+        Ok(Repr {
+            next_header: header.next_header(),
+            frag_offset: header.frag_offset(),
+            more_frags: header.more_frags(),
+            ident: header.ident()
+        })
+    }
+
+    /// Return the length, in bytes, of a header that will be emitted from this high-level
+    /// representation.
+    pub fn buffer_len(&self) -> usize {
+        field::IDENT.end
+    }
+
+    /// Emit a high-level representation into an IPv6 Fragment Header.
+    pub fn emit<T: AsRef<[u8]> + AsMut<[u8]> + ?Sized>(&self, header: &mut Header<&mut T>) {
+        header.set_next_header(self.next_header);
+        header.set_reserved(0);
```

Comment:
I did not see an example of how to handle reserved fields. This is what I came up with.

Reply:
resolved in https://github.com/m-labs/smoltcp/pull/168#discussion_r168937617